frontend/src/routes/settings.ts

```diff
@@ -97,5 +99,11 @@ export const clientAction = async ({ request }: ClientActionFunctionArgs) => {
   }
 
   if (requestedToEndSession(formData)) removeSessionTokenAndSelectedRepo();
+
+  const settingsWithoutApiKey = {
+    ...settings,
+    LLM_API_KEY: LLM_API_KEY ? "SET" : "UNSET",
```

Comment:
Rather than a disallow list, let's make this an allow list. I.e. specifically list out the things we want to pass to posthog.

That'll keep anything new from leaking through without us explicitly adding it